xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -3691,17 +3691,13 @@ bool CVideoPlayer::OpenVideoStream(CDVDStreamInfo& hint, bool reset)
   if (hint.flags & AV_DISPOSITION_ATTACHED_PIC)
     return false;
 
-  // set desired refresh rate
-  if (m_playerOptions.fullscreen && CServiceBroker::GetWinSystem()->GetGfxContext().IsFullScreenRoot() &&
-      hint.fpsrate != 0 && hint.fpsscale != 0)
+  // Set refresh rate, at reset request (playback start)
+  if (reset && hint.fpsrate != 0 && hint.fpsscale != 0)
   {
-    if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(CSettings::SETTING_VIDEOPLAYER_ADJUSTREFRESHRATE) != ADJUST_REFRESHRATE_OFF)
-    {
-      double framerate = DVD_TIME_BASE / CDVDCodecUtils::NormalizeFrameduration((double)DVD_TIME_BASE * hint.fpsscale / hint.fpsrate);
-      RESOLUTION res = CResolutionUtils::ChooseBestResolution(static_cast<float>(framerate), hint.width, hint.height, !hint.stereo_mode.empty());
-      CServiceBroker::GetWinSystem()->GetGfxContext().SetVideoResolution(res, false);
-      m_renderManager.TriggerUpdateResolution(framerate, hint.width, hint.height, hint.stereo_mode);
-    }
+    double framerate =
+        DVD_TIME_BASE / CDVDCodecUtils::NormalizeFrameduration(static_cast<double>(DVD_TIME_BASE) *
+                                                               hint.fpsscale / hint.fpsrate);
+    m_renderManager.UpdateResolution(framerate, hint.width, hint.height, hint.stereo_mode);
```

Comment:
This used to be async. Any reason for having changing it to sync? My guess is that there is a small delay between the start of the stream and the trigger of the update  right?

Reply:
the old code may cause confusion, the first res. switch happens at SetVideoResolution (line 3702) that is called in-sync, after that there is TriggerUpdateResolution that request to change resolution in async way but with same res. info data, not needed since the resolution has already swithed,
there is no functional change only cleanup, this has been intruduced with this PR: https://github.com/xbmc/xbmc/pull/14755

Reply:
Oh, nvm o missed the sync call above.

Reply:
m_renderManager.UpdateResolution must only be called from the render thread. What you are doing here might lead to undefined behaviour on some platforms.

Reply:
That's btw also the reason why CRenderManager::UpdateResolution has no need to be thread safe. So you make it worse in any aspect. If a system may not crash because of calling render functions from a different thread, it may fail because of accessing variables from different thread at the same time.

Reply:
i have reworked, shared variables that was just on UpdateLatencyTweak and m_pRenderer->Update() now are async as before

Reply:
i have reworked, shared variables that was just on UpdateLatencyTweak and m_pRenderer->Update() now are async as before

Reply:
i dont understand what you mean but i am not going to make deep refactors here
if you dont like the fix for playlists, and @enen92 agree i will drop it
and i will kept only the `reset` for DMX_SPECIALID_STREAMCHANGE who must never call the updateresolution